generate_translations.mjs

```diff
@@ -1,42 +1,48 @@
 import fs from 'fs';
 import path from 'path';
+import { exit } from 'process';
+import { spawnSync } from 'child_process';
 
 const translationDir = "joinlemmy-translations/translations/";
 const outDir = "src/shared/translations/";
 const translatorsJsonFile = "lemmy-translations/translators.json";
-const statsFile = "lemmy-instance-stats/stats.json";
-const recommendationsFile = "lemmy-instance-stats/recommended-instances.csv";
+const recommendationsFile = "recommended-instances.json";
+const instanceStatsFile = "src/shared/instance_stats.ts";
 const newsDir = "src/assets/news";
 
 fs.mkdirSync(outDir, { recursive: true });
 
 // Write the stats file
 try {
-  const stats = JSON.parse(fs.readFileSync(statsFile, "utf8"));
-  const recommended_domains = fs.readFileSync(recommendationsFile, "utf8").trim().split(',');
-  console.log(recommended_domains);
-  const recommended = stats.instance_details.filter(i => 
-    recommended_domains.includes(i.domain)
-  );
-  const remaining = stats.instance_details.filter(i => 
-    !recommended_domains.includes(i.domain)
-  );
+  const recommended_instances = JSON.parse(fs.readFileSync(recommendationsFile, "utf8"));
+  var all_recommended = [];
+  for (var k in recommended_instances) {
+    if (k != "exclude") {
+    all_recommended.push(...recommended_instances[k]);
+    }
+  }
+  const run = spawnSync("cargo", 
+  ["run", "--", "--start-instances", all_recommended, "--exclude", recommended_instances.exclude], {
```

Comment:
I moved the crawl into a separate yarn command now, so you explicitly have to run `yarn crawl`. Doing `yarn prebuild:dev` wont do any crawl, so its effectively the same as before (crawl results are directly committed to this repo instead of using a submodule). This should also address your other points, as Rust is only necessary to run a new crawl, just like before. And the output will be identical unless you explicitly run a crawl.

There was a minor problem with the docker build which i fixed.